### ISSUE TITLE
Security: noroot.sh allows any user to use tcpdump and tc

### ISSUE THREAD

**Issue by airtower-luna** [ITSC-Group/autosca-tool#11]
Unconditionally adding CAP_NET_RAW and CAP_NET_ADMIN to `tcpdump` and `tc` means *any* user can use them at will, changing the owning group of the executables does not change that unless the executable bit is removed for "other" (`chmod o-x`). However doing that may have undesirable consequences, e.g. it would prevent unprivileged users from using `tcpdump` to analyze existing PCAP files.

A better approach might be to use `sudo` for the privileged operations and suggest a `sudo` configuration that would allow members of a specific group to run `tcpdump` and `tc` without a password. That way users would have an easy choice to either do the group setup, or enter their password when the script calls `sudo`.

**Comment by ApolloLV** [comment]
Yes, using a limited sudo setup would work as well. I just copied the script from somewhere on github without checking all of its implications. I think the chmod o-x solution might work as well, considering that tcpdump and tc reside in /sbin, which means they are not meant to be executed by non-admin accounts anyways, so restricting access for non-admins should have no noticeable impact.

**Comment by ApolloLV** [comment]
https://gist.github.com/zapstar/3d2ff4f345b43ce7918889053503ef84

**Comment by ApolloLV** [comment]
If we want to go the sudo route, a setup similar to https://askubuntu.com/a/654835 could work.

**Comment by airtower-luna** [comment]
I strongly caution against modifying permissions without explicitly warning the user, and it should definitely not be the default. What's "meant" to be called by non-admins doesn't matter, calling things in `/sbin/` explicitly or adding it to the `PATH` is trivial.

Additionally it's generally bad style to mess with software installed by the distribution. It might break things, maybe for future updates, or be broken by them.

A sudo setup with `NOPASSWD` is what I meant, but it should be optional. If someone prefers to enter their password when sudo is called that should be possible too.
